conda/activate.py

```diff
@@ -546,8 +550,13 @@ def index_of_path(paths, test_path):
             if first_idx is None:
                 first_idx = 0
             else:
-                last_idx = index_of_path(path_list, prefix_dirs[-1])
-                assert last_idx is not None
+                last_idx_idx = len(prefix_dirs) - 1
+                last_idx = None
+                while last_idx is None and last_idx_idx > -1:
+                    last_idx = index_of_path(path_list, prefix_dirs[last_idx_idx])
+                    if last_idx is None:
+                        log.info("Did not find path entry {}".format(prefix_dirs[last_idx_idx]))
```

Comment:
if an entry isn't found here, should we end (``break``) the loop?  What's the difference between last_idx and last_idx_idx?  Can these be named any more descriptively?

Reply:
I'm not sure it matters much. It would subtly change the behaviour of a very broken scenario (which I still do not get).

Reply:
Ah, sorry no, this change was made specifically because some of our PATH entries were missing for some users so we do not want to break when one isn't found, we keep stepping back the entry we look for until (eventually, in a very broken setup) we hit the first entry. I'll change last_idx_idx to prefix_dir_idx if you prefer?

Reply:
sure, that's a little bit clearer